backend/ReservationApp/views.py

```diff
@@ -50,3 +52,24 @@ def post(self, request):
         send_confirmation_email(start_time, user.email, restaurant)
 
         return make_success_response()
+
+class ReservationUpdateStateView(APIView):
+    def put(self, request):
+        try:
+            restaurant_id = request.data['restaurant_id']
+            user_id = request.data['user_id']
```

Comment:
You should be expecting the frontend to supply you with a reservation Id which should be used to search for the reservation.

There can be several reservations with the same user_id and restaurant_id assigned to it, so searching by these parameters wouldn't work. For example, if a user has 3 reservations at the same restaurant at different times, then the user_id and restaurant_id for those reservations would all be the same.